mypy/binder.py

```diff
@@ -122,6 +122,7 @@ def put(self, expr: Expression, typ: Type) -> None:
         if not expr.literal:
             return
         key = expr.literal_hash
+        assert key is not None
```

Comment:
I would prefer to see a descriptive assertion message here, maybe "Internal error: binder tried to put non-literal", ditto for ``get`` and ``cleanse``.